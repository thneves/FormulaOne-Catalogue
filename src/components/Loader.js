import '../styles/components/Loader.scss';

const Loader = () => <div className="box"><div className="Loader" /></div>;

export default Loader;
