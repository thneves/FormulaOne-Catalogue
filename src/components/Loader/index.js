import './styles.scss';

const Loader = () => <div className="box"><div className="Loader" /></div>;

export default Loader;
