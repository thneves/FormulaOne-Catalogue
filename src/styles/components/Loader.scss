.box {
  height: 70vh;
  background-color: black;
  display: flex;
  align-items: center;
  justify-content: center;
}

$Loader-color: #fff;
$Loader-size: 14.6rem;
$Loader-offset: 1.7rem;
$Loader-timing: ease-in-out;

.Loader {
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  max-width: $Loader-size;
  margin-top: $Loader-size/2;
  margin-bottom: $Loader-size/2;

  &::before,
  &::after {
    content: "";
    position: absolute;
    border-radius: 50%;
    animation-duration: 1.8s;
    animation-iteration-count: infinite;
    animation-timing-function: $Loader-timing;
    filter: drop-shadow(0 0 $Loader-offset/2.25 rgba($Loader-color, 0.75));
  }

  &::before {
    width: 100%;
    padding-bottom: 100%;
    box-shadow: inset 0 0 0 $Loader-offset $Loader-color;
    animation-name: pulsA;
  }

  &::after {
    width: calc(100% - #{$Loader-offset} * 2);
    padding-bottom: calc(100% - #{$Loader-offset} * 2);
    box-shadow: 0 0 0 0 $Loader-color;
    animation-name: pulsB;
  }
}

@keyframes pulsA {
  0% {
    box-shadow: inset 0 0 0 $Loader-offset $Loader-color;
    opacity: 1;
  }

  50%,
  100% {
    box-shadow: inset 0 0 0 0 $Loader-color;
    opacity: 0;
  }
}

@keyframes pulsB {
  0%,
  50% {
    box-shadow: 0 0 0 0 $Loader-color;
    opacity: 0;
  }

  100% {
    box-shadow: 0 0 0 $Loader-offset $Loader-color;
    opacity: 1;
  }
}
