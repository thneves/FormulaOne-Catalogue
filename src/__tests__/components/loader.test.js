import React from 'react';
import renderer from 'react-test-renderer';
import Loader from '../../components/Loader';

it('Renders Loader', () => {
  const loader = renderer
    .create(
      <Loader />,
    )
    .toJSON();

  expect(loader).toMatchSnapshot();
});