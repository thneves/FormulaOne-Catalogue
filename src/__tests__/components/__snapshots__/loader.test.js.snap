// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Loader 1`] = `
<div
  className="box"
>
  <div
    className="Loader"
  />
</div>
`;
